improver/wxcode/modal_code.py

```diff
@@ -244,28 +316,484 @@ def process(self, cubes: CubeList) -> Cube:
+
+
+class ModalFromGroupings(BaseModalCategory):
+    """Plugin that creates a modal weather code over a period using a grouping
+    approach. Firstly, a wet and dry grouping is computed. Secondly, for the
+    wet grouping, groupings can be provided, such as, "extreme", "frozen" and "liquid",
+    so that wet weather codes can be grouped further. These groupings can be controlled
+    as follows. Firstly, a day weighting functionality is provided so that daytime
+    hours can be weighted more heavily. A wet bias can also be provided, so that
+    wet codes are given a larger weight as they are considered more impactful.
+    A second categorisation is then available for the wet codes. This is useful when
+    e.g. a period is represented using a variety of frozen precipitation weather codes,
+    so that a frozen precipitation weather code can be diagnosed as an appropriate
+    summary. The ignore intensity option allows light and heavy weather types to be
+    considered together when ascertaining the most common weather type. The final
+    daily symbol will be the most common of the light and heavy input codes of
+    the chosen type.
+
+    The ordering of the codes within the category dictionaries guides which
+    category is selected in the event of the tie with preference given to the lowest
+    index. Incrementing the codes within the category dictionaries from most significant
+    code to least significant code helps to ensure that the most significant code is
+    returned in the event of a tie, if desired.
+
+    Where there are different categories available for night and day, the
+    modal code returned is always a day code, regardless of the times
+    covered by the input files.
+    """
+
+    # Day length set to aid testing.
+    DAY_LENGTH = HOURS_IN_DAY
+
+    def __init__(
+        self,
+        broad_categories: Dict[str, int],
+        wet_categories: Dict[str, int],
+        intensity_categories: Dict[str, int],
+        day_weighting: Optional[int] = 1,
+        day_start: Optional[int] = 6,
+        day_end: Optional[int] = 18,
+        wet_bias: Optional[int] = 1,
+        ignore_intensity: Optional[bool] = False,
+        model_id_attr: Optional[str] = None,
+        record_run_attr: Optional[str] = None,
+    ):
+        """
+        Set up plugin.
+
+        Args:
+            broad_categories:
+                Dictionary defining the broad categories for grouping the weather
+                symbol codes. This is expected to have the keys: "dry" and "wet".
+            wet_categories:
+                Dictionary defining groupings for the wet categories. No specific
+                names for the keys are required. Key and values within the dictionary
+                should both be ordered in terms of descending priority.
+            intensity_categories:
+                Dictionary defining intensity groupings. Values should be ordered in
+                terms of descending priority.
```

Comment:
The detail in this doc-string about the intended order of values in the dictionaries should be added to the CLI doc-string.

Reply:
I've updated the CLI docstring to match the plugin.